dev/WebView2/InteractionTests/WebView2Tests.cs

```diff
@@ -2149,10 +2144,68 @@ public void HtmlDropdownTest()
             }
         }
 
+        [TestMethod]
+        [TestProperty("TestSuite", "D")]
+        public void HiddenThenVisibleTest()
+        {
+            using (var setup = new WebView2TestSetupHelper(new[] { "WebView2 Tests", "navigateToBasicWebView2" }))
+            {
+                ChooseTest("HiddenThenVisibleTest");
+                CompleteTestAndWaitForResult("HiddenThenVisibleTest");
+
+                // Clear the cache so we can find the new webview
+                ElementCache.Clear();
+                var webview = FindElement.ById("MyWebView2");
+                Verify.IsTrue(webview.IsOffscreen == false);
+
+                // Click in the webview to ensure we can interact with it
+                Rectangle bounds = webview.BoundingRectangle;
+                Log.Comment("Bounds = X:{0}, Y:{1}, Width:{2}, Height:{3}", bounds.X, bounds.Y, bounds.Width, bounds.Height);
+                var point = new Point(bounds.X + 20, bounds.Y + 20);
+                Log.Comment("Move mouse to ({0}, {1})", bounds.X + 20, bounds.Y + 20);
+                PointerInput.Move(point);
+
+                PointerInput.Press(PointerButtons.Primary);
+                PointerInput.Release(PointerButtons.Primary);
+                Wait.ForIdle();
+
+                WaitForWebMessageResult("HiddenThenVisibleTest");
```

Comment:
Looks like this tests is doing two types of validation - one via CompleteTestAndWaitForResult(), another via WaitForMessageResult(). Just want to make sure - if the 1st part fails, will the test overall still report failure?

Reply:
Yes, the test should fail and bail on the first time a Verify comes back false, whichever WaitFor... that's inside (or not inside).